File: javatest/omnia/data/cache/CachedTest.java
package omnia.data.cache;

import static com.google.common.truth.Truth.assertThat;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

@RunWith(JUnit4.class)
public class CachedTest {

  @Test
  public void cache_didReturnSimpleCache() {
    assertThat(Cached.cache(Object::new)).isInstanceOf(SimpleCacher.class);
  }
}

File: tests/omnia/data/cache/CachedTest.kt
package omnia.data.cache

import com.google.common.truth.Truth
import omnia.data.cache.Cached.Companion.cache
import org.junit.Test
import org.junit.runner.RunWith
import org.junit.runners.JUnit4

@RunWith(JUnit4::class)
class CachedTest {

  @Test
  fun cache_didReturnSimpleCache() {
    Truth.assertThat(cache<Any> { Object() }).isInstanceOf(SimpleCacher::class.java)
  }
}